content/features/proxy2/configure.md

```diff
@@ -0,0 +1,411 @@
+
+Before you and configure and run the proxy, you need the following:
+
+* Memcached, with proxy features enabled, installed on your system.
+* The proxy route library file downloaded to some path on your file system.
+
+For more information, see [Install the built-in proxy]({{<proxy_base_path>}}install).
+
+## Create a proxy configuration file {#create}
+
+Before you can run the proxy, you need to create a configuration that lists the pools of Memcached servers that you want to use as the proxy's back end, and describes the routing trees that you want the proxy to use.
+
+The configuration file is a Lua script that extends the standard proxy route library. Advanced users can modify the route library itself, or even replace it entirely with their own Lua code. The techniques described on this page focus on using the route library as-is, which covers the majority of Memcached proxy use-cases.
+
+### Overview of the configuration file
+
+The configuration file has four general parts:
+
+1. Optional invocations of global functions to turn on logging or debugging features.
+1. An optional `settings{}` block, which sets various proxy settings to values other than their defaults.
+1. A `pools{}` block, which defines groups of backend Memcached servers that the proxy routes queries to.
+1. One or more `routes{}` blocks, which lay out the rules by which the proxy routes requests to the pools defined in the `pools{}` blocks.
+
+    It's common to have only one `routes{}` block. If you want to have the router handle requests differently depending upon connection context, [you can define multiple `routes{}` blocks](#multiple-trees).
+
+The following sections detail the contents of each of the above parts, with examples.
+
+### Enable logging and debugging
+
+To enable logging and debugging features of the proxy, include any of the following function calls at the top of the configuration file:
+
+* `verbose(1)` has Memcached print additional information to standard output when loading the proxy configuration. This setting also lets you use the `say({{<var>}}TEXT{{</var>}})` function to print <var>TEXT</var> to standard output while loading your custom configuration.
+
+* `debug(1)` turns on a debugging mode that is mainly useful if you are modifying the routing library. This setting also lets you use the `dsay({{<var>}}TEXT{{</var>}})` function to print <var>TEXT</var> to standard output while loading the proxy configuration.
+
+### Adjust proxy settings
+
+To fine-tune various proxy behaviors, include a `settings{}` block in your configuration file. The block contains any number of function calls that set values on the `mcp` object. For the full list of available setting functions, see [General API documentation]({{<legacy_proxy_base_path>}}#general-api-documentation).
+
+For example, the following settings block sets a ten-second connection timeout, and an upper limit of five active requests:
+
+```lua
+settings {
+    mcp.backend_connect_timeout(10)
```

Comment:
routelib is abstracting these calls out (I don't think this even works? heh)

Please compare these routelib snippits with examples from the routelib examples directory: https://github.com/memcached/memcached-proxylibs/blob/main/lib/routelib/examples/basic.lua

you can see the settings block looks a lot more natural.

I'm not sure how to best document this. It might be best to double-document the available settings here.

Reply:
> routelib is abstracting these calls out (I don't think this even works? heh)
> 
> Please compare these routelib snippits with examples from the routelib examples directory: https://github.com/memcached/memcached-proxylibs/blob/main/lib/routelib/examples/basic.lua
> 
> you can see the settings block looks a lot more natural.
> 
> I'm not sure how to best document this. It might be best to double-document the available settings here.

I rewrote this section appropriately, and added a new settings reference, in https://github.com/memcached/memcached.github.io/pull/3/commits/e6f61e933771eab9ce6833b6d2a59523ddf2ab47.